systers_portal/blog/views.py

```diff
@@ -389,3 +410,42 @@ def get_context_data(self, **kwargs):
                                                  slug=self.kwargs['slug'])
         context['tag_type'] = "Resource Type"
         return context
+
+
+class UserPinView(LoginRequiredMixin, ListView):
+    template_name = "blog/post.html"
+    model = Resource
+
+    def post(self, request, resource_slug, slug):
+        if request.method == "POST":
+            community = get_object_or_404(Community, slug=slug)
+            resource = get_object_or_404(Resource, community=community,
+                                         slug=resource_slug)
+            user = SystersUser.objects.get(user=self.request.user)
+            user_pins = UserPins.objects.filter(user=user)
+            if user_pins:
+                user_pins.first().add_pin(resource)
+            else:
+                user_pins = UserPins.objects.create(user=user)
+                user_pins.save()
+                user_pins.add_pin(resource)
+            return JsonResponse({"Done": "done"}, safe=False)
+
+
+class RemovePinView(LoginRequiredMixin, ListView):
+    template_name = "blog/post.html"
+    model = Resource
+
+    def post(self, request, resource_slug, slug):
+        if request.method == "POST":
+            done = "0"
```

Comment:
Could you use a HTTP status code 404 or something instead of this, and show a proper error message

Reply:
Done